lib/stat_tracker.rb

```diff
@@ -185,9 +262,81 @@ def count_of_teams
     # game_teams.csv - game_id, team_id, result, head_coach # DYLAN
     # games.csv - game_id, season 
 
+    def winningest_coach(season)
+      coaches = []
+      @game_teams.map do |game_team|
+        coaches << game_team.head_coach
+        coaches.uniq!
+      end
+      
+      coach_win_percents = []
+      coaches.map do |coach|
+        coach_wins = 0
+        coach_games = 0
+        @game_teams.map do |game_team|
+          if game_team.head_coach == coach && find_game(game_team.game_id).season == season
+            if game_team.result == "WIN"
+              coach_wins += 1
+            end
+            coach_games += 1
+          end
+        end
+        if coach_wins > 0 || coach_games > 0
+          coach_win_percent = (coach_wins.to_f / coach_games.to_f)
+          coach_win_percents << {coach: coach, win_percent: coach_win_percent}
+        else
+          return "Error: no game data for selected season."
+        end
+      end
+      coach_win_percents_sorted = coach_win_percents.sort_by { |coach| coach[:win_percent] }
+      winningest_coaches = []
+      coach_win_percents_sorted.filter_map do |coach_hash|
+        if coach_hash[:win_percent] == coach_win_percents_sorted.last[:win_percent]
+          winningest_coaches << coach_hash[:coach]
+        end
+      end
+      winningest_coaches.join(", ")
+    end
+
   # worst_coach
     # game_teams.csv - game_id, team_id, result, head_coach # DYLAN
-    # games.csv - game_id, season 
+    # games.csv - game_id, season
+
+    def worst_coach(season)
```

Comment:
Dang! This method was waaaay more complicated than I thought it would be. Great job!